### PR TITLE
Gcp account integration

### PR DESCRIPTION
**Pull request by dsuraj221**

### CODE REVIEW

**Comment by jbrmg** [comment]
We need GCP account integration as well in our Terraform templates. Any plans on merging this PR?

**Comment by AmitChotaliya** [comment]
> We need GCP account integration as well in our Terraform templates. Any plans on merging this PR?

Hi @jbrmg,

Thanks for the comment. GCP Integration is still in progress. This will be available latest by 15th June.

File: conformity/resource_gcp_org.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func resourceGCPOrgCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*cloudconformity.Client)
+	// Warning or errors can be collected in a slice type`
+	var diags diag.Diagnostics
+    payload := cloudconformity.OrgPayload{}
```
Comment (AmitChotaliya) [inline conformity/resource_gcp_org.go:73]: Change OrgPayload to GCPOrgPayload

File: docs/guides/gcp/gcp_account_create.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+2. Conformity API Key
+
+#### Step 1
+
+##### Terraform Configuration
+
+1. Create terraform `cloud_formation.tf` on `example/gcp` folder for Cloudformation module, create the file if not existing and add the following values:
+```hcl
+resource "conformity_gcp_account" "gcp" {
+    name            = "MySubscription"
```
Comment (AmitChotaliya) [inline docs/guides/gcp/gcp_account_create.md:22]: Change MySubscription to MyProject

File: docs/guides/gcp/gcp_account_create.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+}
+```
+2. To use Conformity and its resources, add the GCP Access Key, Secret Access Key, and Region where the resources will be created and API Key from Conformity Account to the `terraform.tfvars`. 
+
+3. Create `terraform.tfvars` on `example/GCP` folder and add the following values:
+
+```hcl
+apikey = "CLOUD-CONFORMITY-API-KEY"
+region = "GCP-ACCOUNT-REGION"
```
Comment (AmitChotaliya) [inline docs/guides/gcp/gcp_account_create.md:36]: Conformity Region not GCP Region

**Review by AmitChotaliya — COMMENTED** [review]
Add GCP Org Delete/Update Document with simple comment that it is not possible now.

File: example/gcp/main.tf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+resource "conformity_gcp_account" "gcp" {
+    name            = "MySubscription"
```
Comment (AmitChotaliya) [inline example/gcp/main.tf:2]: My Project

File: example/gcp/provider.tf [hunk 5]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_providers {
+    conformity = {
+//     version = "0.3."
```
Comment (AmitChotaliya) [inline example/gcp/provider.tf:4]: Add terraform registery and version details.

File: example/gcp_organisation/provider.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_providers {
+    conformity = {
+//     version = "0.3."
```
Comment (AmitChotaliya) [inline example/gcp_organisation/provider.tf:4]: Same

File: example/import/account/aws/main.tf [hunk 7]
```diff
@@ -1,5 +1,10 @@
 # conformity_aws_account.aws:
-resource "conformity_aws_account" "aws" {}
+resource "conformity_aws_account" "aws" {
+    name = var.account_tag
```
Comment (AmitChotaliya) [inline example/import/account/aws/main.tf:3]: Revert this changes

File: pkg/cloudconformity/create_gcp_account_test.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	assert.Equal(t, accountID, expectedAccountID)
+}
+
+func TestCreateGCPAccountFail(t *testing.T) {
+	// prepare the test
+	client, ts := createHttpTestClient(t, http.StatusForbidden, errResponse)
+	defer ts.Close()
+
+	// run the code
+	externalID, err := client.CreateGCPAccount(AccountPayload{})
```
Comment (AmitChotaliya) [inline pkg/cloudconformity/create_gcp_account_test.go:30]: I don't it will return externalID

File: pkg/cloudconformity/models.go [hunk 9]
```diff
@@ -162,14 +165,15 @@ type AccountSettings struct {
...
-	CoudType      string                `json:"cloud-type,omitempty"`
-	CloudData     *CloudData            `json:"cloud-data,omitempty"`
-	Settings      *AccountSettings      `json:"settings,omitempty"`
+	Name            string                `json:"name"`
+	Environment     string                `json:"environment"`
+	ManagedGroupId  string                `json:"managed-group-id"`
+	Tags            []string              `json:"tags"`
+	Access          accountAccess         `json:"access,omitempty"`
+	Configuration   *AccountConfiguration `json:"configuration,omitempty"`
+	CoudType        string                `json:"cloud-type,omitempty"`
```
Comment (AmitChotaliya) [inline pkg/cloudconformity/models.go:174]: Typo of CloudType
